Sources/StreamChat/Repositories/MessageRepository.swift

```diff
@@ -71,11 +71,11 @@ class MessageRepository {
                 self?.apiClient.request(endpoint: endpoint) {
                     switch $0 {
                     case let .success(payload):
-                        self?.saveSuccessfullySentMessage(cid: cid, message: payload.message) { message in
-                            if let message = message {
+                        self?.saveSuccessfullySentMessage(cid: cid, message: payload.message) { result in
+                            if let message = result.value {
                                 completion(.success(message))
-                            } else {
-                                completion(.failure(.failedToSendMessage))
+                            } else if let error = result.error {
+                                completion(.failure(.failedToSendMessage(error)))
```

Comment:
we miss a fallback else here or use a switch maybe.

Reply:
Nice catch!